### PR TITLE
Shear plots refactor

### PR DESCRIPTION
**Pull request by sweverett**
Main changes have to do with code to make final tangential shear plot. This is now done through a plotter class in `shear_plots.py` in the `shear-profiles` submodule. Finishes the work from the earlier `shear-profiles` refactor in #48.

This code depends on discussed changes to `make_annular.py` & `annular_jmac.py` so that the output table will include NFW truth information if available.

The plotting code runs successfully in a pipe test w/o the above mentioned truth, but has not been tested with a useable truth file yet. We can update as needed then.

### CODE REVIEW

File: superbit_lensing/shear_profiles/shear_plots.py [hunk 1]
```diff
@@ -2,22 +2,45 @@
 from matplotlib import rc,rcParams
 import matplotlib.pyplot as plt
 from astropy.table import Table
-import pdb
+import pudb
+
+def compute_alpha(nfw, radius, gtan, variance):
```
Comment (mcclearyj) [inline superbit_lensing/shear_profiles/shear_plots.py:7]: I ended up adding a `compute_shear_bias()` function to `annular_jmac.py` in the `resample_redshift` branch I've been working in. The shear bias and its uncertainty are computed therein, and adds those values to the metadata of the file f/k/a `run_name_annular_shear_tab.fits` (now `run_name_shear_profile.fits`?). 

For now, we can merge this into `master` as you've written it, but when we integrate the `redshift_resample` branch, the shear bias and uncertainty values can just be accessed from the table.

File: superbit_lensing/shear_profiles/shear_plots.py [hunk 2]
```diff
@@ -26,77 +49,79 @@ def __init__(self, cat_file, truth_file, pix_scale=0.144):
...
+
+        # in arcsec
+        minrad = rbounds[0]
+        maxrad = rbounds[1]
+
+        cat.sort('midpoint_r') # get in descending order
+
+        radius = self.get_angular_radius(cat['midpoint_r'], arcmin=True)
+
+        gtan = cat['gtan_mean']
```
Comment (mcclearyj) [inline superbit_lensing/shear_profiles/shear_plots.py:90]: I switched the column names to `mean_xxx` and `err_xxx`, I'm sorry. Again, let's merge this in and deal with the column name change afterwards.

File: superbit_lensing/shear_profiles/make_annular_catalog.py [hunk 3]
```diff
@@ -332,8 +332,8 @@ def run(self, overwrite=False, vb=False):
             p = f'{self.run_name}_'
         else:
             p = ''
-        outfile = os.path.join(self.outdir, f'{p}_annular_shear_tab.fits')
-        plotfile = os.path.join(self.outdir, f'{p}_tan_shear.pdf')
+        outfile = os.path.join(self.outdir, f'{p}shear_profile.fits')
```
Comment (mcclearyj) [inline superbit_lensing/shear_profiles/make_annular_catalog.py:335]: Would you be OK with making outfile something like `shear_profile_cat.fits` and the plot `shear_profile_plot.pdf` or at least `outfile = shear_profile.fits`, `plotfile = shear_profile_plot.pdf`? That way, it's obvious that the pdf is a plot of the information contained in the FITS file.

**Review by mcclearyj — CHANGES_REQUESTED** [review]
Added comments to PR

File: superbit_lensing/shear_profiles/make_annular_catalog.py (hunk 3 above)
Comment (sweverett) [reply superbit_lensing/shear_profiles/make_annular_catalog.py:335]: I suppose I wasn't that worried about it as the extension tells you what it is. But I agree that the names aren't great. How about `{run_name}_shear_profile_cat.fits` and `shear_profile.pdf`?

File: superbit_lensing/shear_profiles/shear_plots.py (hunk 1 above)
Comment (sweverett) [reply superbit_lensing/shear_profiles/shear_plots.py:7]: Just realized that `compute_shear_bias()` isn't something I can import yet as it isn't in `master`, so I'll keep it for now as you suggested.

**Pull request merged**